node/resolvers/directives/checkAdminAccess.ts

```diff
@@ -18,6 +20,24 @@ export class CheckAdminAccess extends SchemaDirectiveVisitor {
         clients: { identity },
       } = context
 
+      // send metric to collect data about caller and tokens used in the request
+      const operation = field.astNode?.name?.value ?? context.request.url
+      const metric = new AuthMetric(
+        context.vtex.account,
```

Comment:
🤔 
What happens when a header doesn't exist?

Reply:
If the `context.request.header` structure exists but a specific header (e.g. `'x-vtex-caller'`) is not there, we would get an `undefined` value, which is fine and expected.
On the other hand, if the `context.request.header` structure does not exist at all, we would have a problem (error: `Cannot read properties of undefined (reading 'x-vtex-caller')`), in which case this code would be broken even before this change :)